packages/js/data/src/crud/types.ts

```diff
@@ -147,7 +147,14 @@ export type MapSelectors< Type, ResourceName, ParamType, ReturnType > = {
 export type MapActions< Type, ResourceName, ParamType, ReturnType > = {
 	[ Property in keyof Type as `${ Lowercase<
 		string & Property
-	> }${ Capitalize< string & ResourceName > }` ]: (
-		x: ParamType
-	) => ReturnType;
+	> }${ Capitalize< string & ResourceName > }` ]: Type[ Property ] extends (
+		...args: infer P
+	) => unknown
+		? (
+				...args: [
+					first: ParamType,
+					...rest: P extends [ unknown, ...infer R ] ? R : []
+				]
+		  ) => ReturnType
+		: never;
```

Comment:
Can you explain what exactly is happening here?
Why the need for `first: ParamType`?